addons/website_sale_stock/tests/common.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import TransactionCase
+
+
+class TestWebsiteSaleStockCommon(TransactionCase):
+
+    @classmethod
+    def add_product_qty_to_wh(cls, product_id, qty, loc_id):
+        cls.env['stock.quant'].with_context(inventory_mode=True).create({
+            'product_id': product_id,
+            'inventory_quantity': qty,
+            'location_id': loc_id,
+        }).action_apply_inventory()
+
+    @classmethod
+    def create_storable_product_product(cls, **kwargs):
+        return cls.env['product.product'].create({
+            'name': 'Test',
+            'is_storable': True,
+            'allow_out_of_stock_order': False,
+            'website_published': True,
+            **kwargs,
+        })
```

Comment:
common should extend `ProductCommon` (or `WebsiteSaleCommon`), and extend method `_create_product` ideally.

We could even consider that an override in `WebsiteSaleCommon` would be responsible for adding `'website_published': True` in a method override.